File: Difficulty: Easy/Palindrome Number/palindrome-number.cpp
class Solution {
  public:
    bool isPalindrome(int n) {
      string s=to_string(n);
      int r=0;
      int l=s.size()-1;
      while(r<l){
          if(s[r]=='-'){
              r++;
              continue;
          }
          if(s[l]=='-'){
              l--;
              continue;
          }if(s[r]!=s[l]) {
              return false;
          }
          r++;
          l--;
      }
      return true;
    }
};

File: Difficulty: Easy/Palindrome Number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int n) {
        // code here
        String s =Integer.toString(n);
        int r=0,l=s.length()-1;
        while(r<l){
            if(s.charAt(r)=='-'){
                r++;
                continue;
            }if(s.charAt(l)=='-'){
                l--;
                continue;
            }if(s.charAt(r)!=s.charAt(l)){
                return false;
            }
            r++;
            l--;
        }
        return true;
    }
}